gcp-observability/src/main/java/io/grpc/gcp/observability/GcpObservability.java

```diff
@@ -73,17 +116,104 @@ public void close() {
       if (instance == null) {
         throw new IllegalStateException("GcpObservability already closed!");
       }
+      unRegisterStackDriverExporter();
       LoggingChannelProvider.shutdown();
       LoggingServerProvider.shutdown();
       sink.close();
       instance = null;
     }
   }
 
-  private GcpObservability(Sink sink,
+  private void setProducer(
+      InternalLoggingChannelInterceptor.Factory channelInterceptorFactory,
+      InternalLoggingServerInterceptor.Factory serverInterceptorFactory) {
+    if (config.isEnableCloudLogging()) {
+      clientInterceptors.add(channelInterceptorFactory.create());
+      serverInterceptors.add(serverInterceptorFactory.create());
+    }
+    if (config.isEnableCloudMonitoring()) {
+      clientInterceptors.add(
+          InternalCensusStatsAccessor.getClientInterceptor(true, true, true, true));
+      tracerFactories.add(
+          InternalCensusStatsAccessor.getServerStreamTracerFactory(true, true, true));
+    }
+    if (config.isEnableCloudTracing()) {
+      clientInterceptors.add(InternalCensusTracingAccessor.getClientInterceptor());
+      tracerFactories.add(InternalCensusTracingAccessor.getServerStreamTracerFactory());
+    }
+
+    if (!clientInterceptors.isEmpty()
+        || !serverInterceptors.isEmpty()
+        || !tracerFactories.isEmpty()) {
+      InternalGlobalInterceptors.setInterceptorsTracers(
+          clientInterceptors, serverInterceptors, tracerFactories);
+    }
+  }
+
+  private void registerStackDriverExporter(String projectId) throws IOException {
+    if (config.isEnableCloudMonitoring()) {
+      RpcViews.registerAllGrpcViews();
+      StackdriverStatsConfiguration.Builder statsConfigurationBuilder =
+          StackdriverStatsConfiguration.builder();
+      if (!Strings.isNullOrEmpty(projectId)) {
```

Comment:
I can understand null, do we want to include empty to have the same semantics? 2 Qs

- will the ObservabilityConfig ever return empty string for the project ID?
- if it does do we want special or different semantics for empty project ID?

Reply:
`ObservabilityConfig` will not return empty, if the value does not exist in config it returns `null` instead.
Updated the check.